packages/web3torrent/src/pages/file/File.tsx

```diff
@@ -11,13 +11,15 @@ import {parseMagnetURL} from '../../utils/magnet';
 import torrentStatusChecker from '../../utils/torrent-status-checker';
 import {useInterval} from '../../utils/useInterval';
 import './File.scss';
+import WebTorrentPaidStreamingClient from '../../library/web3torrent-lib';
 import _ from 'lodash';
 
 const getTorrentAndPeersData: (
+  web3torrent: WebTorrentPaidStreamingClient,
   setTorrent: React.Dispatch<React.SetStateAction<Torrent>>,
   setPeers: React.Dispatch<React.SetStateAction<TorrentPeers>>
-) => (torrent: Torrent) => void = (setTorrent, setPeers) => torrent => {
-  const liveTorrent = torrentStatusChecker(torrent, torrent.infoHash);
+) => (torrent: Torrent) => void = (web3torrent, setTorrent, setPeers) => torrent => {
+  const liveTorrent = torrentStatusChecker(web3torrent)(torrent, torrent.infoHash);
```

Comment:
```suggestion
  const liveTorrent = torrentStatusChecker(web3Torrent)(torrent, torrent.infoHash);
```

Reply:
does `torrentStatusChecker` need to be curried?

Reply:
no, it isn't, changing it to an extra parameter